File: Python/Program/GraphicsCard.py
from Komponen import Komponen

class GraphicsCard(Komponen):
    def __init__(self, vram_gb=0, gpu_model="", cuda_cores=0, base_clock=0, boost_clock=0, memory_type="", memory_bandwidth=0, cooling_type="", merk="", nama=""):
        super().__init__(merk, nama)
        self.vram_gb = vram_gb
        self.gpu_model = gpu_model
        self.cuda_cores = cuda_cores
        self.base_clock = base_clock
        self.boost_clock = boost_clock
        self.memory_type = memory_type
        self.memory_bandwidth = memory_bandwidth
        self.cooling_type = cooling_type

    def set_vram_gb(self, vram_gb):
        self.vram_gb = vram_gb

    def get_vram_gb(self):
        return self.vram_gb

    def set_gpu_model(self, gpu_model):
        self.gpu_model = gpu_model

    def get_gpu_model(self):
        return self.gpu_model

    def set_cuda_cores(self, cuda_cores):
        self.cuda_cores = cuda_cores

    def get_cuda_cores(self):
        return self.cuda_cores

    def set_base_clock(self, base_clock):
        self.base_clock = base_clock

    def get_base_clock(self):
        return self.base_clock

    def set_boost_clock(self, boost_clock):
        self.boost_clock = boost_clock

    def get_boost_clock(self):
        return self.boost_clock

    def set_memory_type(self, memory_type):
        self.memory_type = memory_type

    def get_memory_type(self):
        return self.memory_type

    def set_memory_bandwidth(self, memory_bandwidth):
        self.memory_bandwidth = memory_bandwidth

    def get_memory_bandwidth(self):
        return self.memory_bandwidth

    def set_cooling_type(self, cooling_type):
        self.cooling_type = cooling_type

    def get_cooling_type(self):
        return self.cooling_type

File: Java/Program/GraphicsCard.java
public class GraphicsCard extends Komponen {
    private int vramGB;
    private String gpuModel;
    private int cudaCores;
    private int baseClock;
    private int boostClock;
    private String memoryType;
    private int memoryBandwidth;
    private String coolingType;

    public GraphicsCard() {
        super();
    }

    public GraphicsCard(int vramGB, String gpuModel, int cudaCores, int baseClock, int boostClock, String memoryType, int memoryBandwidth, String coolingType, String merk, String nama) {
        super(merk, nama);
        this.vramGB = vramGB;
        this.gpuModel = gpuModel;
        this.cudaCores = cudaCores;
        this.baseClock = baseClock;
        this.boostClock = boostClock;
        this.memoryType = memoryType;
        this.memoryBandwidth = memoryBandwidth;
        this.coolingType = coolingType;
    }

    public void setVramGB(int vramGB) {
        this.vramGB = vramGB;
    }

    public int getVramGB() {
        return this.vramGB;
    }

    public void setGpuModel(String gpuModel) {
        this.gpuModel = gpuModel;
    }

    public String getGpuModel() {
        return this.gpuModel;
    }

    public void setCudaCores(int cudaCores) {
        this.cudaCores = cudaCores;
    }

    public int getCudaCores() {
        return this.cudaCores;
    }

    public void setBaseClock(int baseClock) {
        this.baseClock = baseClock;
    }

    public int getBaseClock() {
        return this.baseClock;
    }

    public void setBoostClock(int boostClock) {
        this.boostClock = boostClock;
    }

    public int getBoostClock() {
        return this.boostClock;
    }

    public void setMemoryType(String memoryType) {
        this.memoryType = memoryType;
    }

    public String getMemoryType() {
        return this.memoryType;
    }

    public void setMemoryBandwidth(int memoryBandwidth) {
        this.memoryBandwidth = memoryBandwidth;
    }

    public int getMemoryBandwidth() {
        return this.memoryBandwidth;
    }

    public void setCoolingType(String coolingType) {
        this.coolingType = coolingType;
    }

    public String getCoolingType() {
        return this.coolingType;
    }
}